nav2_dynamic_params/include/nav2_dynamic_params/dynamic_params_client.hpp

```diff
@@ -90,24 +90,36 @@ class DynamicParamsClient
     if (*full_path.begin() != '/') {
       full_path = '/' + full_path;
     }
-    auto client = std::make_shared<rclcpp::SyncParametersClient>(node_, full_path);
+
     std::vector<rclcpp::Parameter> params;
-    if (client->wait_for_service(100ms)) {
+
+    if (full_path == join_path(node_->get_namespace(), node_->get_name())) {
```

Comment:
This function is starting to get long and complicated. I'd break it up into smaller sub functions.